install/helm/agones/templates/controller.yaml

```diff
@@ -60,6 +60,15 @@ spec:
         release: {{ .Release.Name }}
         heritage: {{ .Release.Service }}
     spec:
+      {{- if .Values.agones.controller.topologySpreadConstraints }}
```

Comment:
My suggestion, let's mimic was we did for `agones.controller.affinity` - which was by default have `{}` and let people add what they want.

That way we don't have to maintain the structure of `topologySpreadConstraints` over time.

https://github.com/googleforgames/agones/blob/eb4035e1f78e7a8e0c147a013a5405f904ab9899/install/helm/agones/templates/controller.yaml#L63-L70

@roberthbailey any objections?

Reply:
I'm a bit on the fence. 

Right now there isn't an example of how to use `agones.controller.affinity` in `values.yaml` even though it's listed as a valid field on https://agones.dev/site/docs/installation/install-agones/helm/ which I think is OK because we should be encouraging using dedicated nodes (and they can't be set at the same time). 

Do you think we will consider recommending setting default topology spread constraints? It seems like it has the potential to graduate into a best practice, in which case we'd want to set it in values.yaml to turn it on by default.

Reply:
Having a chat on the community meeting - decision was made:

Allow an arbitrary block under `topologySpreadConstraints` like we do for our `affinity` values.

So let's not define each element, let's allow a user to provide any of the values they need in a `.Values.agones.controller.topologySpreadConstraints` helm variable.

Reply:
Do we want to update it as follows?

```
{{- if .Values.agones.controller.topologySpreadConstraints }}
topologySpreadConstraints:
  {{- toYaml .Values.agones.controller.topologySpreadConstraints | nindent 4 }}
{{- end }}
```